plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -310,6 +310,29 @@ func (sl *SocketListener) Start(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func udpListen(network string, address string) (net.PacketConn, error) {
+	switch network {
+	case "udp", "udp4", "udp6":
+		var addr *net.UDPAddr
+		var err error
+		var ifi *net.Interface
+		if spl := strings.SplitN(address, "%", 2); len(spl) == 2 {
+			address = spl[0]
+			ifi, err = net.InterfaceByName(spl[1])
+			if err != nil {
+				return nil, err
+			}
+		}
+		addr, err = net.ResolveUDPAddr(network, address)
+		if err == nil {
+			if addr.IP.IsMulticast() {
+				return net.ListenMulticastUDP(network, ifi, addr)
+			}
+		}
+	}
+	return net.ListenPacket(network, address)
```

Comment:
Let's use ListenUDP when the addr is not multicast, that could save us the second dns resolution.  Still need ListenPacket for unixgram and ip.

Reply:
Good call. Updated to resolve this.